### DISCUSSION TITLE
HTML content for labels

### DISCUSSION THREAD

**Discussion by stockiNail** [chartjs/chartjs-plugin-annotation#695] [Ideas]
Pretty often the users ask to add HTML content on the chart by the annotation plugin.

Currently this is possible only using `svg` and `foreignObject` elements creating an image and setting the content as image element.
This workaround has got some constraints, like:

1. styles can not leverage of imported CSS definitions but must be defined in style attributes (as far as I have seen)
2. links can not be clickable

Starting from this status, I was thinking if make sense to define label content (in label, box and line annotations) as HTMLElement or HTML strings.
The element should be added in the DOM tree as child of the canvas and move to the right position at drawing time.

In this way, you can not use all options defined now for labels. For instance `drawTime` will be ignored because the label will be shown always on top of the canavs (`drawTime: 'afterDraw'`).
I'm not 100% convinced but this new feature.

What do you think?

**Comment by LeeLenaleee** [comment]
Don't think it is a way this plugin should go.
Mostly because I think you might open Pandora's box with potential issues like why does this component not render or why does it render incorrectly. Or does it support custom jsx components for example.

In my opinion if people really want it, it should be a separate plugin
